vyper/semantics/analysis/imports.py

```diff
@@ -219,6 +214,20 @@ def _load_import_helper(
         search_paths = self.input_bundle.search_paths.copy()  # noqa: F841
         raise ModuleNotFound(module_str, hint=hint) from err
 
+    def _load_file(self, path: PathLike, level: int) -> CompilerInput:
+        ast = self.graph.current_module
+        current_search_path = Path(ast.resolved_path).parent
+
+        search_paths = self.absolute_search_paths if level == 0 else [current_search_path]
+
+        with self.input_bundle.temporary_search_paths(search_paths):
+            res = self.input_bundle.load_file(path)
+
+        if level != 0:
+            self.input_bundle.search_paths += [current_search_path]
```

Comment:
```suggestion
            self.input_bundle.search_paths.append(current_search_path)
```

Reply:
this is kind of weird actually because on line 221, we already set search_paths to `[current_search_path]`

Reply:
I think the issue is we need to set it temporarily to only one path, but then we need to add it to search paths in case there is an absolute import that uses it

Reply:
isn't that handled in the recursion by line 221?
https://github.com/vyperlang/vyper/pull/4268/files/c91f13e870bce8001eeec83dd96b8320c53d4343#diff-dc7dbf75948a3fce0db2cfa0fa819c964a534604d06f72ae88d18e3e3eef78b6R221

Reply:
i pushed a clean up in a140226e89d803fdf3d254b3400f2af165238134

Reply:
I though that example such as the following should work (but checked with python and you are right, it shouldn't).
```
#a.py
from .subdir import b
```

and 

```
#b.py
from subdir.subsubdir import c
```